### PR TITLE
chore(starter): add `.npmrc`

### PR DESCRIPTION
**Pull request by Barbapapazes**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

In documentation, you speak a lot about using `nitro` with `pnpm`. But when installing deps with `pnpm`, `h3` utils are just any since `h3` is not a first party dep and user must manually install it or add a `.npmrc`. I think that added, by default this file, could help.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Hebilicious** [comment]
In the main branch, h3 is in the dependencies.

I see you made this PR against the starter branch though.
Are you trying to make this change for the presets starters? We have this repo here https://github.com/unjs/nitro-preset-starter

**Comment by Barbapapazes** [comment]
> In the main branch, h3 is in the dependencies.
> 
> I see you made this PR against the starter branch though.
> Are you trying to make this change for the presets starters? We have this repo here https://github.com/unjs/nitro-preset-starter

I want to make this PR to the starter, the one that is downloaded when doing npx giget nitro

File: .npmrc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+shamefully-hoist=true
```
Comment (pi0) [inline .npmrc:1]: Nitro is compatible with pnpm hoist. And we shall not enable this flag to make sure it keeps compatible.

**Comment by pi0** [comment]
Thanks for the pull request. The top level `.npmrc` is not for starter template but for repository itself and for the reason of making sure being pnpm compatible, we should not enable this flag.

Adding such npmrc to the starter template might be a good idea in the meantime until solving "type issue" for sub dependencies.

**Comment by pi0** [comment]
Sorry my bad! Let's have this fix for starter until resolving type issue.

**Pull request merged**